packages/xod-client/src/editor/components/inspectorWidgets/index.js

```diff
@@ -42,47 +42,57 @@ const submitOnEnter = {
   },
 };
 
-export const WIDGET_MAPPING = {
-  [SELECTION_ENTITY_TYPE.NODE]: {
-    [WIDGET_TYPE.BOOLEAN]: {
-      component: BoolWidget,
-      props: {
-        type: PIN_TYPE.BOOLEAN,
-        commitOnChange: true,
-      },
+const WIDGET_MAPPING = {
+  [WIDGET_TYPE.BOOLEAN]: {
+    component: BoolWidget,
+    props: {
+      type: PIN_TYPE.BOOLEAN,
+      commitOnChange: true,
     },
-    [WIDGET_TYPE.NUMBER]: {
-      component: NumberWidget,
-      props: {
-        type: PIN_TYPE.NUMBER,
-        keyDownHandlers: R.merge(widgetNumberKeysDownHandlers, submitOnEnter),
-      },
+  },
+  [WIDGET_TYPE.NUMBER]: {
+    component: NumberWidget,
+    props: {
+      type: PIN_TYPE.NUMBER,
+      keyDownHandlers: R.merge(widgetNumberKeysDownHandlers, submitOnEnter),
     },
-    [WIDGET_TYPE.STRING]: {
-      component: StringWidget,
-      props: {
-        type: PIN_TYPE.STRING,
-        keyDownHandlers: submitOnEnter,
-      },
+  },
+  [WIDGET_TYPE.STRING]: {
+    component: StringWidget,
+    props: {
+      type: PIN_TYPE.STRING,
+      keyDownHandlers: submitOnEnter,
     },
-    [WIDGET_TYPE.PULSE]: {
-      component: PulseWidget,
-      props: {
-        type: PIN_TYPE.PULSE,
-        commitOnChange: true,
-      },
+  },
+  [WIDGET_TYPE.PULSE]: {
+    component: PulseWidget,
+    props: {
+      type: PIN_TYPE.PULSE,
+      commitOnChange: true,
     },
-    [WIDGET_TYPE.TEXTAREA]: {
-      component: DescriptionWidget,
-      props: { type: 'string' },
+  },
+  [WIDGET_TYPE.TEXTAREA]: {
+    component: DescriptionWidget,
+    props: { type: 'string' },
+  },
+  [WIDGET_TYPE.DEAD]: {
+    component: DeadPinWidget,
+    props: {
+      type: PIN_TYPE.DEAD,
     },
-    [WIDGET_TYPE.DEAD]: {
+  },
+};
+
+export const getNodeWidgetConfig = type => {
+  if (isGenericPinType(type)) {
+    return {
       component: DeadPinWidget,
```

Comment:
Naming bore: not sure using the *dead* pin widget is a good idea for generics. Better generalize it to something like `DisabledInputWidget`.